CHANGELOG.md

```diff
@@ -1,3 +1,185 @@
+ * A new `ClosureReason::LocallyCoopClosedUnfundedChannel` was added (#3881).
+ * Some arguments to `ChannelManager::pay_for_offer[_from_human_readable_name]`
+   have moved behind `optional_params` (#3808, #3903).
+ * `Event::PaymentSent::bolt12_invoice` was added for proof-of-payment (#3593).
+ * Channel values are now synchronized via RGS, improving scoring (#3924).
+ * `SendOnlyMessageHandler` was added, implemented for `ChainMonitor`, and
+   an instance added to `MessageHandler`. Note that `ChainMonitor` does not yet
+   send any messages, though will in the future (#3922).
+ * `lightning_background_processor::NO_{ONION_MESSENGER,LIQUIDITY_MANAGER}` were
+   added to simplify background processor init without some args (#4100, #4132).
+ * `ChannelManager::set_current_config` was added (#4038).
+ * Onion messages received to a blinded path we generated are now authenticated
+   implicitly rather than explicitly in blinded path `Context`s (#3917, #4144).
+ * `OMNameResolver::expire_pending_resolution` has been added for those who
+   cannot or do not wish to call `new_best_block` regularly (#3900).
+ * `lightning-liquidity`'s LSPS1 client now supports BOLT 12 payments (#3649).
+ * `LengthReadable::read` has been renamed `read_from_fixed_length_buffer` and
+   is implemented for all `Readable` (#3579).
+ * `LengthReadable` is now required to read various objects which consume the
+   full available buffer (#3640).
+ * structs in `lightning-liquidity` were renamed to be globally unique (#3583).
+ * Renamed `SpendableOutputDescriptor::outpoint` to `spendable_outpoint` (#3634)
+
+## Performance Improvements
+ * `ChainMonitor::load_existing_monitor` was added and should be used on startup
+   to load existing `ChannelMonitor`s rather than via `Persist`, avoiding
+   re-persisting each `ChannelMonitor` during startup (#3996).
+ * RGS data application was further sped up (#3581).
+
+## Bug Fixes
+ * `FilesystemStore::list` is now more robust against race conditions with
+   simultaneous `write`/`archive` operations (#3799).
+ * Pending async persistence of `ChannelMonitorUpdate`s required to forward an
+   HTLC can no longer result in the HTLC being forgotten if the channel is
+   force-closed (#3989).
+ * `lightning-liquidity`'s service support now properly responds to the
+   `ListProtocols` message (#3785).
+ * A rare race which might lead `PeerManager` (and `lightning-net-tokio`) to
+   stop reading from a peer until a new message is sent to that peer has been
+   fixed (#4168).
+ * The fields in `SocketAddress::OnionV3` are now corectly parsed, and the
+   `Display` for such addresses is now lowercase (#4090).
+ * `PeerManager` is now more conservative about disconnecting peers which aren't
+   responding to pings in a timely manner. This may reduce disconnections
+   marginally when forwarding gossip to a slow peer (#4093, #4096).
+ * `BlindedMessagePath::new_with_dummy_hops` was added (but is not used by
+   default, #3726).
```

Comment:
It looks like we do use `new_with_dummy_hops` by default for non-compact BOLT 12 structs. Also, is this really a bug fix?

Reply:
> Also, is this really a bug fix?

Nope! I think I started to put it there when I thought it was used by default, which I might call a bugfix, but it is not :(

Reply:
> > It looks like we do use new_with_dummy_hops by default for non-compact BOLT 12 structs.
> 
> 
> 
> Indeed, but its only called in `lightning::onion_message::messenger` from `create_blinded_paths_from_iter` which is only called with the `compact` argument set to `false` by `NodeIdMessageRouter`, not `DefaultMessageRouter`. `NodeIdMessageRouter` is not used by default anywhere AFAICT.

Ah okay, I thought we did non-compact for invoice requests and invoices but never mind